presentation/virtualKeyboard.brs

```diff
@@ -48,8 +86,30 @@ Function virtualKeyboard_Initialize(msgPort As Object, userVariables As Object,
           ?"hide: ";m.virtualKeyboard.hide()
         endif
         return true
+      
+      ' Only load the keyboard widget after html widget has finished loading
+      else if type(event) = "roHtmlWidgetEvent" then
+        data = event.GetData()
+        if type(data) = "roAssociativeArray" then
+          if data.reason = "load-finished"
+            m.virtualKeyboard = invalid
+            ?"load-finished test"
+            m.virtualKeyboard = virtualKeyboard_Setup(m.rectangle, m.transform, m.msgPort)
+            return false
+          endif
+        endif
       endif
       return false
     End Function
   }
 End Function
+
+Function virtualKeyboard_Setup(rectangle, transform, msgPort)
+
+  virtualKeyboard = createObject("roVirtualKeyboard", rectangle)
+  virtualKeyboard.SetTransform(transform)
+  virtualKeyboard.setResource("file:///virtualKeyboard/bsvirtualkb.html")
```

Comment:
The file may be unexpectedly present. Would be helpful to add a check for file existence.